[EDITS]Copy to edit (#1244)
mulesoft/docs-connectors: edifact-edi/2.2/modules/ROOT/edits/edifact-edi-connector-config-topics.adoc

= EDIFACT EDI Connector 2.2 - Configuring Schemas - Mule 4

> 

Reviewer: edit request

this content is very similar to the X12 docs you just edited for me. I have done my best to incorporate those edits into the EDIFACT docs. Type: CE, Scope: all except lines 25-29 in index.adoc, Due: two days from start is fine.

> If your implementation conventions differ from the standard, you can create schemas to describe your messages. Standard versions of the schemas for all defined EDIFACT messages and supported versions are included with this connector.

Reviewer: Should these sentences be reversed?

== Set Your Schema Locations

> You can configure schema locations in the Anypoint Studio XML view.

Reviewer: Omit: it's in the task steps.
```suggestion
```

To configure schemas in Anypoint Studio:

> . In Anypoint Studio, switch to the XML view by clicking **Configuration XML**.

Reviewer: To avoid repeating "in Anypoint Studio":
```suggestion
. Switch to the XML view by clicking *Configuration XML*.
```

. Modify your EDIFACT EDI configuration to include a list of all schemas by adding an `+<http://edischema[edi:schema]>+` element for each document type:

[source,xml,linenums]
----
<edifact-edi:config name="EDIFACT_EDI__Configuration" identKeys="true" doc:name="EDIFACT EDI: Configuration">
  <edifact-edi:schemas>
    <edifact:schema value="/edifact/d96a/ORDERS.esl"/>
  </edifact-edi:schemas>
</edifact-edi:config>
----

> After you create a global element in Studio for your EDIFACT EDI, configure any schemas and operations.

Reviewer: To me, this just sounds better? Your choice:
```suggestion
After you create a global element for your EDIFACT EDI in Studio, configure any schemas and operations.
```

=== Configuration Options in XML

All values listed in the Studio configuration can be set directly in XML.

General parameters control both Send and Receive document handling.

[%header%autowidth.spread]
|===

> |XML Value |Studio Option

Reviewer: Should these columns be reversed, given the intro. that puts Studio config. first and XML second (Line 28)?

Reviewer: If you switch these, do so GLOBALLY in this file.

> |schemas=list of values |List of schema paths the connector can use. These paths can be either a file system or `classpath`.

Reviewer: ```suggestion
|schemas=list of values |List of schema paths the connector can use. Each path can be either a file system or `classpath`.
```

|stringSubstitutionChar |Substitution character used to replace invalid characters within string values when using the syntax level UNOA or UNOB.

> |separatorUsage="USE_DEFAULTS" "USE_SPECIFIED_FOR_WRITES" "USE_SPECIFIED_FOR_ALL" |Use specified separator or release characters as defaults for messages.

Reviewer: parallelism among column texts:
```suggestion
|separatorUsage="USE_DEFAULTS" "USE_SPECIFIED_FOR_WRITES" "USE_SPECIFIED_FOR_ALL" |Specification to use either specified separator or release characters as defaults for messages.
```

|dataSeparator="+" |Data element separator character.
|componentSeparator=":" |Component element separator character.
|repetitionSeparator="*" |Repetition separator character.
|segmentTerminator="'" |Segment terminator character.
|releaseCharacter="?" |Release character.
|===

> == Self Identification Parameters

Reviewer: ```suggestion
== Self-Identification Parameters
```

> Use these parameters to identify your side of the trading partner relationship.

Reviewer: ```suggestion
Use these parameters to identify your side of the trading partner relationship:
```

[%header%autowidth.spread]
|===
|XML Value |Studio Option
|interchangeIdSelf |Mule interchange Sender or Receiver ID (UNB2.1/UNB3.1).

> |interchangeIdQualifierSelf |Interchange Sender or receiver ID qualifier (UNB2.2/UNB3.2).

Reviewer: ```suggestion
|interchangeIdQualifierSelf |Interchange Sender or Receiver ID qualifier (UNB2.2/UNB3.2).
```

|===

== Partner Identification Parameters

These identify your partner.

[%header%autowidth.spread]
|===
|XML Value |Studio Option
|interchangeIdPartner |Partner interchange Sender or Receiver ID for Mule application (UNB2.1/UNB3.1).
|interchangeIdQualifierPartner |Partner interchange Sender or Receiver ID qualifier (UNB2.2/UNB3.2).
|===

== Parser Parameters for Error Conditions

These control the parser operation and the types of error conditions that reject Receive messages.

[%header%autowidth.spread]
|===
|XML Value (default) |Studio Option
|enforceLengthLimits="true" |Enforce the minimum and maximum lengths for receive values.
|enforceValueRepeats="true" |Enforce the repetition count limits for receive values.

> |enforceReadCharacters="true" |Enforce valid characters for a Read. Only applies when using syntax levels `UNOA` and `UNOB`.

Reviewer: ```suggestion
|enforceReadCharacters="true" |Enforce valid characters for a Read operation. Only applies when using syntax levels `UNOA` and `UNOB`.
```

|allowUnknownSegments="false" |Allow unknown segments in a message.
|enforceSegmentOrder="true" |Enforce the segment order in a message.
|allowUnusedSegments="false" |Allow segments marked as `Unused` in a message.
|enforceSegmentRepeats="true" |Enforce segment repetition count limits in a message.
|receiveEncodingOverride="false" |Java character encoding used for Receive messages. This overrides receive UNB.

> |requireUniqueInterchanges="true" |Require unique Interchange Control References (such as UNB.5). The default behavior records the interchange numbers previously processed and rejects duplicate interchange numbers from the same partner. It is derived from the interchange Sender and Receiver identification. Setting to `false` enables received interchange processing to continue and relies on the application flow to track the numbers and manage them appropriately.

Reviewer: ```suggestion
|requireUniqueInterchanges="true" |Require unique interchange control references (such as UNB.5). The default behavior records the interchange numbers previously processed and rejects duplicate interchange numbers from the same partner. It is derived from the interchange Sender and Receiver identification. Setting to `false` enables received interchange processing to continue and relies on the application flow to track the numbers and manage them appropriately.
```

Reviewer: Evaluate whether this item and the one that follows need to have so much information in their description (compared to the others). Perhaps these details are covered elsewhere?

> |requireUniqueMessages="false" |Enforce globally unique Message Reference Numbers (UNH.1) for received messages. By default, message reference numbers need to be unique only within a particular interchange. Setting to `true` requires message reference numbers to be unique across all interchanges received from the same partner and application. This is derived from the interchange sender and recipient identification.

Reviewer: ```suggestion
|requireUniqueMessages="false" |Enforce globally unique message reference numbers (UNH.1) for received messages. By default, message reference numbers need to be unique only within a particular interchange. Setting to `true` requires message reference numbers to be unique across all interchanges received from the same partner and application. This is derived from the interchange sender and recipient identification.
```

|daysToStore="30" |Minimum number of days to store interchange, group, and message numbers for uniqueness checking.
|===

== Writer Parameters

These control the types of writer operations.

[%header%autowidth.spread]
|===
|XML Value |Studio Option
|characterEncoding="ASCII_B" |Character encoding used for writing messages. There must be one defined in each EDIFACT syntax version.
|sendSyntaxVersion="VERSION4" |Syntax version for Send messages. Used to determine the data structures for service segments and CONTRL acknowledgment message structures in DataSense metadata for both Send and Receive.

> |lineEnding="NONE" "LF" "CRLF" "CR" |Line ending to add between segments. Improves the readability of the output message text by adding line endings between segments.

Reviewer: Repetitive; merge all into one sentence?

|alwaysSendUNA="false" |Always send the UNA service segment. A UNA is included only in an interchange when the syntax separator characters don't match the defaults for both the syntax version and character encoding when in use. Setting to `true` means a UNA is always sent.

> |enforceWriteCharacters="true" |Enforce valid characters for write. This applies when using syntax levels `UNOA` and `UNOB`.

Reviewer: ```suggestion
|enforceWriteCharacters="true" |Enforce valid characters for a Write operation. This applies when using syntax levels `UNOA` and `UNOB`.
```

Reviewer: GLOBALLY lowercase message reference numbers and interchange control reference.

|sendUniqueMessageNumbers="false" |Send unique Message Reference Numbers (UNH.1). `false` (default) assigns message reference numbers sequentially within each interchange and reuses them in a different interchange. `true` assigns unique transaction set numbers across all interchanges sent to the same partner. It is derived by the interchange sender and recipient identification.
|initialInterchangeReference="1" |The initial Interchange Control Reference used for outgoing messages.
|initialMessageReference="1" |The initial Message Reference Number used for outgoing messages.
|requestAcks="false" |Request acknowledgments for Send interchanges using the Acknowledgment Request field (UNB.9)
|testIndicator="" |Test indicator digit used on Send interchanges (UNB.11). By default interchanges are not sent as tests.
|useSuppliedValues="false" |Choose your own data values for control-segment identifiers such as `UNB` and `UNZ` segments or use the supplied values. `false` always generates control numbers when writing letting you choose values.
|===

== Next Step

After you complete configuring the connector, you can try
the xref:edifact-edi-connector-examples.adoc[Examples].

== See Also

> * https://help.mulesoft.com[MuleSoft Help Center]

Reviewer: BREAK here

Reviewer: RESUME
